src/backend/gporca/libgpopt/src/operators/CPhysicalHashJoin.cpp

```diff
@@ -385,24 +399,15 @@ CPhysicalHashJoin::PdshashedMatching(
 									 ulSourceChild);
 		}
 	}
-	if (pdrgpexpr->Size() != ulDlvrdSize)
+	if (pdrgpexpr->Size() != ulDlvrdSize ||
```

Comment:
And this condition is generally redundant, since it completely repeats the previous one. Therefore, I propose to merge this branch with the previous one.